### ISSUE TITLE
Detection works very well but confidence is very low

### ISSUE THREAD

**Issue by ash1995** [avanetten/simrdwn#24]
I've attached the inference results using the simrdwn. As you can see, the detection is quite good, but I'm not sure why the confidence is so low. Strangely this is true for a training image as well which doesn't make sense. Do you have any ideas as to what could be a possible reason for this?

![atlanta_nadir7_catid_1030010003d22f00_732701_3731889_thresh 0 05](https://user-images.githubusercontent.com/32295036/51970378-35527e00-2490-11e9-8b38-226434246f54.jpg)
![atlanta_nadir16_catid_1030010002649200_748451_3735939_thresh 0 05](https://user-images.githubusercontent.com/32295036/51970379-35527e00-2490-11e9-866b-8148b8471036.jpg)
![atlanta_nadir27_catid_1030010003472200_734501_3743589_thresh 0 05](https://user-images.githubusercontent.com/32295036/51971035-b65e4500-2491-11e9-8e11-ef6c2b8a779c.jpg)

**Comment by avanetten** [comment]
Glad you got it working.  The confidences have always been relatively low for the YOLT models, though increasing training time will increase confidence levels.

**Comment by ash1995** [comment]
I see. Initially I trained the model for 50,000 iterations. But even after doubling the training time (100,000 iterations) the model confidence is still quite low. 

![atlanta_nadir14_catid_10300100039ab000_736301_3730089_thresh 0 05](https://user-images.githubusercontent.com/32295036/52172908-ea3bb200-2792-11e9-97fc-065ebc06f9d3.jpg)
![atlanta_nadir10_catid_1030010003caf100_742151_3722439_thresh 0 05](https://user-images.githubusercontent.com/32295036/52172915-03dcf980-2793-11e9-9a81-60574d9386fc.jpg)

On a side note,
I tried one of the tensorflow models as well to see how well it does (framework:**SSD**, num of batches:30,000, batch_size:16). The results are clearly not as good as the YOLT models. 

![atlanta_nadir10_catid_1030010003caf100_732701_3725139_thresh 0 05](https://user-images.githubusercontent.com/32295036/52173082-0725b480-2796-11e9-8bf1-51a20a3204b1.jpg)

While this could be due to not training for enough iterations, could it be due to the usage of the ssd config under the **_orig** config? I'm not sure what the _orig and _altered_v0 configs for the tensorflow models represent but am I correct in assuming that the **altered** configs are those optimised for satellite images and the originals are the original configs as specified in their respective papers?

**Comment by ash1995** [comment]
I trained the YOLT model even longer and so far the model confidence hasn't improved. From it's results it's obviously a very strong detector but I'm unable to quantify how good the model is.

**YOLT detection**

![atlanta_nadir27_catid_1030010003472200_734501_3740889_thresh 0 05](https://user-images.githubusercontent.com/32295036/52765062-0285b980-303c-11e9-9956-f1f1b623fed2.jpg)

In most cases, the YOLT model does correct and incorrect detection with almost similar confidence scores. That is it would correctly detect a building with 5% confidence and incorrectly detect a building with 6% confidence which makes it, as I mentioned before, hard to determine how good it actually is. I suppose we could just not use confidence score as a metric and have the model output all detections and see which are correct. Would like your input on the matter.

Regarding the other tensorflow models, as mentioned in the above post, it's not as good as the yolt model (at least in the case of SSD). 

**SSD Detection**

Plot threshold strength = 0.3
![atlanta_nadir27_catid_1030010003472200_734501_3740889_thresh 0 3](https://user-images.githubusercontent.com/32295036/52765955-1979db00-303f-11e9-9611-0f4edb5b9ea5.jpg)

Plot threshold strength = 0.2
![atlanta_nadir27_catid_1030010003472200_734501_3740889_thresh 0 2](https://user-images.githubusercontent.com/32295036/52766060-6b226580-303f-11e9-996e-38880124b45c.jpg)


In summary, 
1. Should confidence not be considered for YOLT models?
2. Are there some recommended strategies to train the tensorflow models such that they perform better? Perhaps choice of parameters?

Thanks

**Comment by avanetten** [comment]
I'm not sure what's causing the confidences to be so low.  You could try retraining with the recently updated v2 and see if the problem persists.

**Comment by ash1995** [comment]
I recently tried out recent version of YOLT and the problem still persists as you can see from the images below. Note that I've only tried training the new model for 30000 epochs.

![Atlanta_nadir10_catid_1030010003CAF100_732701_3730989_thresh=0 05](https://user-images.githubusercontent.com/32295036/56888288-a6e89b00-6a84-11e9-957d-e4448d5f5677.jpg)
![Atlanta_nadir10_catid_1030010003CAF100_733151_3741789_thresh=0 05](https://user-images.githubusercontent.com/32295036/56888289-a6e89b00-6a84-11e9-9eae-f5ff020e03d1.jpg)

I'll update this post after trying out a couple more experiments i.e. longer training time, tf models etc.

**Comment by ryanh18** [comment]
I'm having a similar issue training on my custom data from the darknet source code. I've bounced back & forth between it being anchor values, batch size, stronger GPU, different baseline weights, Yolo vs Yolo-tiny... any ideas or solutions you had here?
